src/core/src/core_logic/PatchAssessor.py

```diff
@@ -27,14 +27,15 @@ def __init__(self, env_layer, execution_config, composite_logger, telemetry_writ
 
         self.composite_logger = composite_logger
         self.telemetry_writer = telemetry_writer
+        self.is_telemetry_setup = False
         self.status_handler = status_handler
 
         self.package_manager = package_manager
 
     def start_assessment(self):
         """ Start an update assessment """
         self.status_handler.set_current_operation(Constants.ASSESSMENT)
-        self.telemetry_setup()
+        self.telemetry_writer.is_agent_compatible()
```

Comment:
This should be a condition check that returns true or false; if false, assessment writes it's own status entry (error) into status handler, and raises an exception to stop the run.

if not self.telemetry_writer.is_agent_compatible():
     #write out the error and raise exception as you had before (lines 81 to 83).